File: src/shared/types/note.ts
// 改动：Note / NoteDraft 增加可选 cover 字段（封面图片 URL）
export type NoteTone = 'neutral' | 'danger' | 'primary'

export type NotesView = 'all' | 'favorites' | 'trash' | 'folders'

/**
 * 标签为 note-scoped：嵌在笔记 tags[] 内，不是全局标签实体。
 * 筛选条候选项通过对各笔记 tags 聚合得到。
 */
export interface NoteTag {
  id: string
  name: string
  /** 视觉 tone 可选；当前 UI 统一默认色，不再按 tone 分色。 */
  tone?: NoteTone
}

export interface Note {
  id: string
  title: string
  /**
   * 编辑器内容。
   * 新笔记为 ProseMirror doc 的 JSON 字符串化结果（见 noteDomain.EMPTY_DOC_JSON）。
   * 历史纯文本/HTML 字符串在渲染时由 RichEditor 做 fallback 解析。
   */
  content: string
  excerpt: string
  /** note-scoped 标签列表（非全局标签表）。 */
  tags: NoteTag[]
  folderId: string | null
  isFavorite: boolean
  isDeleted: boolean
  /** ISO 时间；进入废纸篓时写入，恢复时清空。 */
  deletedAt: string | null
  /** 封面图片 URL；null/undefined 表示无封面。 */
  cover?: string | null
  /** 是否置顶（在列表顶部优先显示）。 */
  pinned?: boolean
  /** 是否只读（锁定内容防止意外修改）。 */
  readOnly?: boolean
  createdAt: string
  updatedAt: string
}

export interface NoteDraft {
  title: string
  content: string
  tags?: NoteTag[]
  folderId?: string | null
  cover?: string | null
}

export interface NotesFilter {
  view: NotesView
  query: string
  tagId: string | null
}

File: backend/app/schemas/note.py
"""笔记与文件夹相关 schema。"""
from pydantic import BaseModel, Field


class NoteTagSchema(BaseModel):
    id: str
    name: str
    tone: str | None = None


class NoteDraftRequest(BaseModel):
    title: str = Field("", max_length=500)
    content: str = Field("", description="ProseMirror doc JSON 字符串")
    tags: list[NoteTagSchema] | None = None
    folderId: str | None = None
    cover: str | None = Field(None, max_length=512)


class NoteUpdateRequest(BaseModel):
    title: str | None = Field(None, max_length=500)
    content: str | None = None
    tags: list[NoteTagSchema] | None = None
    folderId: str | None = None
    isFavorite: bool | None = None
    isDeleted: bool | None = None
    deletedAt: str | None = None
    cover: str | None = Field(None, max_length=512)
    pinned: bool | None = None
    readOnly: bool | None = None


class FolderDraftRequest(BaseModel):
    name: str = Field(..., min_length=1, max_length=64)
    icon: str = Field("folder", max_length=32)
    parentId: str | None = None


class FolderUpdateRequest(BaseModel):
    name: str | None = Field(None, min_length=1, max_length=64)
    icon: str | None = Field(None, max_length=32)
    parentId: str | None = None


class FolderBatchDeleteRequest(BaseModel):
    ids: list[str]
